src/components/KTableView/KTableView.vue

```diff
@@ -542,36 +536,36 @@ const hasColumnVisibilityMenu = computed((): boolean => {
   }
 
   // show when not loading and there is data
-  return !props.loading && !!props.data && !!props.data.length
+  return !loading && !!data && !!data.length
 })
-const columnVisibilityDisabled = computed((): boolean => props.loading || !(props.data && props.data.length))
+const columnVisibilityDisabled = computed((): boolean => loading || !(data && data.length))
 // columns whose visibility can be toggled
-const visibilityColumns = computed((): TableViewHeader[] => tableHeaders.value.filter((header: TableViewHeader) => header.hidable && !isSpecialColumn(header.key)))
+const visibilityColumns = computed(() => tableHeaders.value.filter((header) => header.hidable && !isSpecialColumn(header.key)))
 // visibility preferences from the host app (initialized by app)
-const visibilityPreferences = computed((): Record<string, boolean> => hasColumnVisibilityMenu.value ? props.tablePreferences.columnVisibility || {} : {})
+const visibilityPreferences = computed(() => (hasColumnVisibilityMenu.value ? tablePreferences.columnVisibility || {} : {})) as Ref<ColumnVisibility>
 // current column visibility state
-const columnVisibility = ref<Record<string, boolean>>(hasColumnVisibilityMenu.value ? props.tablePreferences.columnVisibility || {} : {})
+const columnVisibility = ref(hasColumnVisibilityMenu.value ? tablePreferences.columnVisibility || {} : {}) as Ref<ColumnVisibility>
 
 const tableWrapperHeight = ref<string>('100%')
 const isScrollableVertically = ref<boolean>(false)
 const isScrolledVertically = ref<boolean>(false)
 const isScrolledHorizontally = ref<boolean>(false)
 const isScrollableRight = ref<boolean>(false)
-const sortColumnKey = ref('')
+const sortColumnKey = ref('') as Ref<ColumnKey | ''>
```

Comment:
Why do we need to cast rather than type? (Some question for a few refs in the PR)